dali/operators/audio/nonsilence_op.h

```diff
@@ -0,0 +1,250 @@
+                  args.cutoff_db = -enclosing_->cutoff_db_[sample_id];
+                  args.reference_db = enclosing_->reference_db_[sample_id];
+                  args.reference_max = enclosing_->reference_max_[sample_id];
+                  args.window_length = enclosing_->window_length_[sample_id];
+                  args.reset_interval = enclosing_->reset_interval_[sample_id];
+
+                  auto res = DetectNonsilenceRegion(sample_id, args);
+                  auto beg_ptr = output_begin[sample_id].mutable_data<detail::OutputType>();
+                  auto len_ptr = output_length[sample_id].mutable_data<detail::OutputType>();
+                  *beg_ptr = res.first;
+                  *len_ptr = res.second;
+              });
+    }
+    tp.WaitForWork();
+  }
+
+
+  /**
+   * Performs nonsilent region detection for single sample
+   * @return (begin_idx, length)
+   */
+  template<typename InputType>
+  std::pair<int, int>
+  DetectNonsilenceRegion(int sample_id, const Args<InputType> &args) {
+    SetupKernel();
+    RunKernel(sample_id, args.input, {args.window_length, args.reset_interval});
+    auto dbs = view_as_tensor<float>(intermediate_buffers_[sample_id]);
+    kernels::signal::DecibelCalculator<float> dbc(10.f, args.reference_max ? max(dbs)
+                                                                           : args.reference_db);
+    return LeadTrailThresh(make_cspan(dbs.data, dbs.num_elements()), dbc.db2signal(args.cutoff_db));
+  }
+
+
+  void SetupKernel() {
+    intermediate_buffers_.resize(nsamples_);
+  }
+
+
+  template<typename InputType>
+  void
+  RunKernel(int sample_id, TensorView<StorageCPU, const InputType, 1> in, const MmsArgs &mms_args) {
+    kernels::KernelContext kctx;
+    kernels::signal::MovingMeanSquareCpu<InputType> mms;
+    auto reqs = mms.Setup(kctx, in, mms_args);
+    intermediate_buffers_[sample_id].Resize(reqs.output_shapes[0][sample_id]);
+    auto out = view_as_tensor<float>(intermediate_buffers_[sample_id]);
+    mms.Run(kctx, out.template to_static<1>(), in, mms_args);
+  }
+
+
+  /**
+   * @brief Performs leading and trailing thresholding.
+   *
+   * Returns index of a first value above the threshold in the buffer and
+   * the length up to the point, where last value above the threshold appears in the buffer.
+   *
+   * Example:
+   * buffer: [0, 0, 0, 0, 50, 50, 0, 0]
+   * cutoff: 20
+   * return: {4, 2}
+   *
+   * @return (begin_idx, length)
+   */
+  template<typename T>
+  static std::pair<int, int> LeadTrailThresh(span<const T> buffer, T cutoff) {
+    assert(buffer.size() > 0);
+    int begin = -1;
+    int end = buffer.size();
+    while (begin < end && buffer[++begin] < cutoff);  // NOLINT
+    if (begin == end) return {-1, 0};
+    while (buffer[--end] < cutoff);  // NOLINT
+    return {begin, end - begin + 1};
+  }
+
+
+  template<typename T>
+  T max(TensorView<StorageCPU, T, DynamicDimensions> tv) {
+    T max = -std::numeric_limits<T>::infinity();
```

Comment:
```suggestion
    T max = std::numeric_limits<T>::min();
```

Reply:
Actually, neither xD `::lowest()` is proper here...

From https://en.cppreference.com/w/cpp/types/numeric_limits
```
type	lowest()	min()		max()
 
uchar	0		0		255
int	-2147483648	-2147483648	2147483647
float	-3.40282e+38	1.17549e-38	3.40282e+38
double	-1.79769e+308	2.22507e-308	1.79769e+308
```